### ISSUE TITLE
Unsure if I have the right use case

### ISSUE THREAD

**Issue by jmazar** [msu-coinlab/pymoo#147]
Hello,

I'm currently working with a problem where I have a dataset where I have 7500 data points and 3 features. I'm looking to find the pareto set of possible combinations of my 3 features, and then make a decision of what to use based on the set. I believe that pymoo (or something like it) could be used to solve this issue, but I can't quite figure out how. 

I've come up with a single, basic optimization function and one constraint. However I'm not how/where/if it is possible to pass my dataset to be optimized over. I'm trying with just two of the features to see if I can get something working, but I don't think I'm getting a good result. It takes a few seconds per iteration to run and in the end it only gives me one datapoint in the scatterplot. I feel like I'm doing something incredibly wrong. Would appreciate any help!

```
data # 7500,2 matrix

class MyProblem(Problem):
    def __init__(self, dataset):
        super().__init__(n_var=2,
                         n_obj=1,
                         n_constr=1,
                         xl=np.array([-800, -2]),
                         xu=np.array([2800, 1]))

        self.dataset = dataset

    def _evaluate(self, X, out, *args, **kwargs):
        f1 = self.dataset[:, 0] + self.dataset[:, 1]

        g1 = self.dataset[:, 0] # only values less than 0 are valid

        out["F"] = f1
        out["G"] = g1


problem = MyProblem(data)

algorithm = NSGA2(pop_size=7500)

res = minimize(problem, algorithm, ("n_gen", 5), verbose=True, seed=1)

plot = Scatter()
plot.add(res.F, color="red")
plot.show()

```

**Comment by julesy89** [comment]
Optimizing over a data set can be done the way you do it. Pass the `data` to the constructor and use it in the `_evaluate` function.

Solving a single-objective problem will only result in one data point. There is no second objective to be in trade-off with.

**Comment by jmazar** [comment]
Right, that makes sense. I thought my two features would be on the X and Y axis but I understand now.

Would it be feasible to just have each feature be its own optimization function? Or if I get to more than 3 or so is that unreasonable?

**Comment by julesy89** [comment]
You can use two objectives, where one input is directly mapped to an objective. 

```
f1 = ...
f2 = X[:, feature]
out["F"] = np.column_stack([f1, f2])
```

However, not knowing your optimization problem and just looking at your code snippet I have some doubts if you have already clearly formulated your optimization problem.
Also, keep in mind NSGA2 is known to not scale well with an increasing number of objectives. I recommend using NSGA3 for three or more.

**Comment by jmazar** [comment]
Hahaha, I also have doubts that I clearly formulated my optimization problem. A bit more explanation - I have generated a dataset via simulations and I have added various features/metadata that I think correlate with better out of sample performance. It is difficult to come up with a cost function because it is difficult to calculate the value of a feature compared to another. Bringing them into the same unit space where I can quantify tradeoff is not something I know how to do. 

I just know I want to calculate the pareto set of my features that dominate the rest, then I can make a choice from that.

Thank you for the advice re: NSGA3. I'll close this now as I believe you've answered my question. Thank you again.
